src/autopas/selectors/FeatureVectorEncoder.h

```diff
@@ -0,0 +1,205 @@
+    for (const auto &option : _newton3Options) {
+      if (vec[pos++] == 1.) {
+        if (newton3) {
+          utils::ExceptionHandler::exception(
+              "FeatureVectorEncoder.oneHotDecode: Vector encodes more than one newton3. (More than one value for "
+              "newton3 "
+              "equals 1.)");
+        }
+        newton3 = option;
+      }
+    }
+    if (not newton3) {
+      utils::ExceptionHandler::exception(
+          "FeatureVectorEncoder.oneHotDecode: Vector encodes no newton3. (All values for newton3 equal 0.)");
+    }
+
+    const auto &[container, traversal, estimator] = *cteOption;
+    return FeatureVector(container, cellSizeFactor, traversal, estimator, *dataLayout, *newton3);
+  }
+
+  /**
+   * Convert Feature vector to cluster representation for GaussianCluster.
+   * Discrete values are encoded using their index in given std::vector.
+   * @param vec vector to encode
+   * @return cluster encoded vector
+   */
+  [[nodiscard]] std::pair<Eigen::VectorXi, Eigen::VectorXd> convertToCluster(const FeatureVector &vec) const {
+    int containerTraversalEstimatorIndex = static_cast<int>(
+        std::distance(_containerTraversalEstimatorOptions.begin(),
+                      std::find(_containerTraversalEstimatorOptions.begin(), _containerTraversalEstimatorOptions.end(),
+                                std::make_tuple(vec.container, vec.traversal, vec.loadEstimator))));
+    int dataLayoutIndex = static_cast<int>(std::distance(
+        _dataLayoutOptions.begin(), std::find(_dataLayoutOptions.begin(), _dataLayoutOptions.end(), vec.dataLayout)));
+    int newton3Index = static_cast<int>(
+        std::distance(_newton3Options.begin(), std::find(_newton3Options.begin(), _newton3Options.end(), vec.newton3)));
+
+    Eigen::Vector3i vecDiscrete({containerTraversalEstimatorIndex, dataLayoutIndex, newton3Index});
+    Eigen::VectorXd vecContinuous(FeatureVector::featureSpaceContinuousDims);
+    vecContinuous << vec.cellSizeFactor;
+    return std::make_pair(vecDiscrete, vecContinuous);
+  }
+
+  /**
+   * Inverse of convertToCluster. Convert cluster representation back
+   * to Feature vector.
+   * @param vec cluster encoded vector
+   * @return decoded vector
+   */
+  FeatureVector convertFromCluster(const std::pair<Eigen::VectorXi, Eigen::VectorXd> &vec) {
+    const auto &[vecDiscrete, vecContinuous] = vec;
+    auto [container, traversal, estimator] = _containerTraversalEstimatorOptions[vecDiscrete[0]];
+
+    return FeatureVector(container, vecContinuous[0], traversal, estimator, _dataLayoutOptions[vecDiscrete[1]],
+                         _newton3Options[vecDiscrete[2]]);
+  }
+
+ private:
+  std::vector<FeatureVector::ContainerTraversalEstimatorOption> _containerTraversalEstimatorOptions;
+  std::vector<DataLayoutOption> _dataLayoutOptions;
+  std::vector<Newton3Option> _newton3Options;
+
+  /**
+   * Dimensions of a one-hot-encoded vector
+   */
+  size_t _oneHotDims;
```

Comment:
```suggestion
  size_t _oneHotDims{0};
```
because you do not init this in the constructors